include/ion/node.h

```diff
@@ -121,6 +123,11 @@ class Node {
     Port oport(const std::string& pn);
     std::vector<std::tuple<std::string, Port>> oports() const;
 
+    std::vector<std::tuple<std::string, Port>> dynamic_iports() const;
+    std::vector<std::tuple<std::string, Port>> dynamic_oports() const;
+
+    void  validate_port_address_alignment ()const ;
```

Comment:
This function looks detecting data hazard. It's better to be more intuitive naming
 https://en.wikipedia.org/wiki/Hazard_(computer_architecture)